src/App/Fossa/ManualDeps.hs

```diff
@@ -80,17 +82,39 @@ analyzeFossaDepsFile ::
   , Has Exec sig m
   ) =>
   Path Abs Dir ->
+  Maybe FilePath ->
   Maybe ApiOpts ->
   VendoredDependencyOptions ->
   m (Maybe SourceUnit)
-analyzeFossaDepsFile root maybeApiOpts vendoredDepsOptions = do
-  maybeDepsFile <- findFossaDepsFile root
+analyzeFossaDepsFile root maybeCustomFossaDepsPath maybeApiOpts vendoredDepsOptions = do
+  maybeDepsFile <-
+    case maybeCustomFossaDepsPath of
+      Nothing -> findFossaDepsFile root
+      Just filePath -> retrieveCustomFossaDepsFile filePath
   case maybeDepsFile of
     Nothing -> pure Nothing
     Just depsFile -> do
       manualDeps <- context "Reading fossa-deps file" $ readFoundDeps depsFile
       context "Converting fossa-deps to partial API payload" $ Just <$> toSourceUnit root depsFile manualDeps maybeApiOpts vendoredDepsOptions
 
+retrieveCustomFossaDepsFile ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has ReadFS sig m
+  ) =>
+  FilePath ->
+  m (Maybe FoundDepsFile)
+retrieveCustomFossaDepsFile fossaDepsPath = do
+  let extension = takeExtension fossaDepsPath
+  file <- (validateFile fossaDepsPath)
+
+  if extension == ".yml" || extension == ".yaml"
+    then pure $ Just $ ManualYaml file
+    else
+      if extension == ".json"
+        then pure $ Just $ ManualJSON file
+        else pure Nothing
```

Comment:
We should fail fatally here - since we are silently ignoring, if user has provided some non compliant file. I think for `--config`, we fail fatally. Prefer if replicate behaviour here: `fatalText "expected ... but, received: "`